citrination_client/views/tests/test_model_template_builder.py

```diff
@@ -0,0 +1,160 @@
+import json
+import requests_mock
+import os
+import time
+import uuid
+
+from citrination_client.views.data_view_builder import DataViewBuilder
+from citrination_client.views.search_template.client import SearchTemplateClient
+
+from citrination_client.views.model_template.client import ModelTemplateClient
+
+from citrination_client.views.client import DataViewsClient
+
+
+def load_file_as_json(path):
+    """
+    Given a filepath, loads the file as a dictionary from JSON
+
+    :param path: The path to a JSON file
+    """
+    with open(path, "r") as f:
+        parsed_dict = json.load(f)
+    return parsed_dict
+
+
+def test_workflow():
+    print('Starting data view creation workflow')
+
+    site = "https://citrination.com"
+    search_template_client = SearchTemplateClient(os.environ["CITRINATION_API_KEY"], site)
+    model_template_client = ModelTemplateClient(os.environ["CITRINATION_API_KEY"], site)
+    data_views_client = DataViewsClient(os.environ["CITRINATION_API_KEY"], site)
+
+    search_template_url = "{}/api/search_templates{}"
+    datasets_url = "{}/api/datasets{}"
+    data_view_url = "{}/api/data_views{}"
+    descriptors_url = "{}/descriptors{}"
+
+    with requests_mock.Mocker() as m:
+        # Setup mocks
+        m.post(
+            search_template_url.format(site, '/from-dataset-ids'),
+            json=dict(data=load_file_as_json('./citrination_client/views/tests/test_search_template.json'))
+        )
+
+        m.post(
+            search_template_url.format(site, '/prune-to-extract-as'),
+            json=dict(data=load_file_as_json('./citrination_client/views/tests/test_search_template.json'))
+        )
+
+        m.post(
+            datasets_url.format(site, '/get-available-columns'),
+            json=dict(data=load_file_as_json('./citrination_client/views/tests/available_columns.json'))
+        )
+
+        m.post(
+            descriptors_url.format(site, '/trigger-job'),
+            json=dict(data=dict(poll_url=descriptors_url.format(site, '/job-status/1234')))
+        )
+
+        m.post(
+            descriptors_url.format(site, '/job-status'),
+            json=dict(data=load_file_as_json('./citrination_client/views/tests/column_descriptors.json'))
+        )
+
+        m.post(
+            data_view_url.format(site, ''),
+            json=dict(id=555)
+        )
+
+        # Get available columns
+        available_columns = search_template_client.get_available_columns([1234])
+        assert len(available_columns) == 524
+
+        # Create a search template from dataset ids
+        search_template = search_template_client.create([1234], available_columns)
+        assert search_template['query'][0]['system'][0]['tags'][0]['category'] == 'General'
+
+        # Create ML configuration
+        dv_builder = DataViewBuilder()
+        dv_builder.add_real_descriptor('Property Melting point', 'Output', 'Infinity', '0')
```

Comment:
This is actually not crazy. Maybe I take it my prior comment about unifying this back. OR DO BOTH!?